### PR TITLE
[atari] Optimize findfreeiocb

### PR DESCRIPTION
**Pull request by pfusik**

### CODE REVIEW

File: mos-platform/atari8-common/findfreeiocb.s [hunk 1]
```diff
@@ -16,16 +16,11 @@
...
         beq     found
-        txa
-        clc
-        adc     #$10
-        tax
-        cmp     #$80
-        bcc     loop
-        inx                     ; return ZF cleared
+        adc     #$10            ; C = 0
+        bpl     loop
```
Comment (mysterymath) [inline mos-platform/atari8-common/findfreeiocb.s:25]: This should have a comment explaining why `bpl` gives the loop termination condition, now that it's no longer an explicit compare against `#$80`.

Comment (pfusik) [reply mos-platform/atari8-common/findfreeiocb.s:25]: What's your point? Checking the N flag is the 6502's idiomatic way to determine $00-$7F vs $80-$FF. It's the `cmp #$80` that looked strange.
Also, all Atari programmers should be aware that there are 8 IOCBs.
I'm okay with adding a comment, but I don't know what should it explain?

Comment (mysterymath) [reply mos-platform/atari8-common/findfreeiocb.s:25]: It's not actually checking whether the value is negative, it's literally checking whether it is >= 0x80. It's looping through the IOCBs, each of which are 0x10 bytes long, and there are 8 of them.

**Review by mysterymath — APPROVED** [review]
LGTM!

**Pull request merged**